### PR TITLE
Bring in Chef 16 and PJC that can build on Ruby 2.7

### PR DESCRIPTION
**Pull request by tyler-ball**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/gems/Gemfile [hunk 1]
```diff
@@ -82,7 +82,7 @@ group(:omnibus_package) do
 
   # knife plugins
   gem "knife-acl", ">= 1.0.6"
```
Comment (tas50) [inline components/gems/Gemfile:84]: knife-acl is built into chef 16 so this one can go away

File: components/gems/Gemfile [hunk 2]
```diff
@@ -101,7 +101,7 @@ group(:omnibus_package) do
   gem "chef-sugar"
```
Comment (tas50) [inline components/gems/Gemfile:101]: We should consider removing this since we ship this in chef 15.X more or less

File: components/gems/Gemfile [hunk 3]
```diff
@@ -48,8 +48,8 @@ group(:omnibus_package) do
   # gems to Rubygems now, so letting this float on latest should always give us the latest
   # stable release. May have to re-pin around major version bumping time, or during patch
   # fixes.
-  gem "chef", "= 15.10.12 "
-  gem "chef-bin", "= 15.10.12 "
+  gem "chef", "= 16.0.287 "
+  gem "chef-bin", "= 16.0.287 "
   gem "ohai", ">= 15"
```
Comment (tas50) [inline components/gems/Gemfile:53]: This should be > 16 just in case some other dep tries to pull it down

File: components/gems/Gemfile [hunk 4]
```diff
@@ -48,8 +48,8 @@ group(:omnibus_package) do
...
   gem "cheffish", ">= 14.0.1"
```
Comment (tas50) [inline components/gems/Gemfile:54]: Might as well make this 15+ now

File: components/gems/Gemfile.lock [hunk 5]
```diff
@@ -321,7 +325,7 @@ GEM
       concurrent-ruby (~> 1.0)
       ffi-yajl (~> 2.2)
       http (~> 4)
-    chef-utils (15.10.12)
+    chef-utils (16.0.287)
     chef-vault (4.0.1)
```
Comment (tas50) [inline components/gems/Gemfile.lock:329]: There's a chef-vault dep in the gemfile that can go away now that it's a dep of chef itself

File: components/gems/Gemfile (hunk 2 above)
Comment (btm) [reply components/gems/Gemfile:101]: to clarify what @tas50, we've put functionality in chef-utils, which ships in core chef, that is mostly equivalent to chef sugar starting around 15.5+

**Pull request merged**